storybookjs/storybook
# New docs versioning - repo proposal

Issue: related to #1773

## What I did
I've added a functionality to the docs-v2 build script that will commit a new exported docs version to a separate repo

## Tasks TBD
- [x] Fix relative to root links
- [x] Fix dynamic chunks are not being downloaded for the version site.
- [x] Use/Create Storybook bot user to commit to the docs repo.
- [x] Create storybook docs repo
- [x] Add the docs repo to the CI
- [x] Add `DOCS_REPO` variable with access token to CI
- [x] Automate docs deployment in main storybook CI
- [x] Generate the `versions.json` for a versions drop-down
- [x] Fix edge case when the same exported version is generated with a different build hash (because maybe something was changed but version is still the same) : when copying to the main `_next` dir the old app version data should be deleted.

## Example
https://storypuke-mlglazhotd.now.sh
https://storypuke-mlglazhotd.now.sh/3-0-0
https://storypuke-mlglazhotd.now.sh/3-0-1
https://storypuke-mlglazhotd.now.sh/3-2-5
https://storypuke-mlglazhotd.now.sh/3-2-9

## How to test
Meanwhile:
1. Create a repo 
2. Add a basic `package.json` - it's needed because the build script is looking for it to upgrade the version.
3. Create authentication token from here : https://github.com/settings/tokens
4. In `docs-v2/package.json` change the build script to this: `"build": "cross-env DOCS_REPO=https://{your_user_here}:{token_from_step_3}@github.com/{repo} node scripts/build.js",`
5. Run the `build` script from the step 4
6. Change the version in the `docs-v2/package.json`
7. Run the `build` script from the step 4 again to see that newer version committed to the repo as well
8. Optional step - host on cloud - look the `now` configuration here: https://github.com/igor-dv/storypuke-docs/blob/master/package.json
